src/ast.ml

```diff
@@ -22,7 +22,8 @@ type inline_element =
   | `Styled of style * inline_element with_location list
   | `Reference of
     reference_kind * string with_location * inline_element with_location list
-  | `Link of string * inline_element with_location list ]
+  | `Link of string * inline_element with_location list
+  | `Math of bool * string ]
```

Comment:
I'm not a fan of the bool to distinguish between the two forms. I think it's not necessary to recognize `{%math:` specifically, `` `Math `` should only correspond to the `{m` syntax.

Reply:
I'm also not a fan of bools, though I do think this should be used for both display and inline math (I think @julow's comment refers to the older idea of reusing `{%math:...}` that we've since ruled out). Could we have a `type math_style = [ `Inline | `Display ]` instead?

Reply:
I think the distinction between `{m ...}` and `{math ...}` is that one is inline and the other is a block. These should be two different constructors in the AST: one in `inline_element` and a different one in `nestable_block_element`. I propose to name them `` `Math_span `` and `` `Math_block `` to reuse the naming of code spans and code blocks.

Reply:
Yep, I'm going to go with the separated `` `Math_span`` and `` `Math_block` I think, it does make sense to construct it symmetrically with code spans and blocks

Reply:
Makes sense to me!